scanpy/tools/_rank_genes_groups.py

```diff
@@ -135,317 +462,63 @@ def rank_genes_groups(
             groups_order = [str(n) for n in groups_order]
         if reference != 'rest' and reference not in set(groups_order):
             groups_order += [reference]
-    if (
-        reference != 'rest'
-        and reference not in set(adata.obs[groupby].cat.categories)
-    ):
+    if reference != 'rest' and reference not in set(adata.obs[groupby].cat.categories):
```

Comment:
```suggestion
    if reference != 'rest' and reference not in adata.obs[groupby]:
```

I realize this wasn't changed in this PR, but this should already be efficient for categorical dtypes.

Reply:
Set is not needed, but your suggestion doesn't seem to work properly.

Reply:
Ah, I must have tested it for the array not the series. Apparently `__contains__` for a series looks in the index. Just removing the `set` would be good.